### PR TITLE
Fix configuration of Add location button Visibility

### PR DESCRIPTION
**Pull request by aurangzaibumer**
**IMPORTANT: Where possible all PRs must be linked to a Github issue**

Fixes https://github.com/onaio/fhir-resources/issues/3155

image

**Engineer Checklist**
- [ ] I have written **Unit tests** for any new feature(s) and edge cases for bug fixes
- [ ] I have added any strings visible on UI components to the `strings.xml` file
- [ ] I have updated the  [CHANGELOG.md](./CHANGELOG.md) file for any notable changes to the codebase
- [ ] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the project's style guide
- [ ] I have built and run the FHIRCore app to verify my change fixes the issue and/or does not break the app 
- [ ] I have checked that this PR does NOT introduce **breaking changes** that require an update to **_Content_** and/or **_Configs_**? _If it does add a sample here or a link to exactly what changes need to be made to the content._

**Code Reviewer Checklist**
- [ ] I have verified **Unit tests** have been written for any new feature(s) and edge cases
- [ ] I have verified any strings visible on UI components are in the `strings.xml` file
- [ ] I have verifed the [CHANGELOG.md](./CHANGELOG.md) file has any notable changes to the codebase
- [ ] I have verified the solution has been implemented in a configurable and generic way for reuseable components
- [ ] I have built and run the FHIRCore app to verify the change fixes the issue and/or does not break the app

### CODE REVIEW

File: android/geowidget/src/main/java/org/smartregister/fhircore/geowidget/screens/GeoWidgetFragment.kt [hunk 1]
```diff
@@ -76,8 +76,9 @@ class GeoWidgetFragment : Fragment() {
     }
   internal var useGpsOnAddingLocation: Boolean = false
   internal var mapLayers: List<MapLayerConfig> = ArrayList()
-  internal var shouldLocationButtonShow: Boolean = true
+  internal var showCurrentLocationButton: Boolean = true
   internal var shouldPlaneSwitcherButtonShow: Boolean = true
```
Comment (dubdabasoduba) [inline android/geowidget/src/main/java/org/smartregister/fhircore/geowidget/screens/GeoWidgetFragment.kt:80]:
```suggestion
  internal var showPlaneSwitcherButton: Boolean = true
```

File: android/geowidget/src/main/java/org/smartregister/fhircore/geowidget/screens/GeoWidgetFragment.kt [hunk 2]
```diff
@@ -333,8 +335,9 @@ class Builder {
     }
   private var useGpsOnAddingLocation: Boolean = false
   private var mapLayers: List<MapLayerConfig> = ArrayList()
-  private var shouldLocationButtonShow: Boolean = true
+  private var showCurrentLocationButton: Boolean = true
   private var shouldPlaneSwitcherButtonShow: Boolean = true
```
Comment (dubdabasoduba) [inline android/geowidget/src/main/java/org/smartregister/fhircore/geowidget/screens/GeoWidgetFragment.kt:339]:
```suggestion
  private var showPlaneSwitcherButton: Boolean = true
```

File: android/geowidget/src/main/java/org/smartregister/fhircore/geowidget/screens/GeoWidgetFragment.kt [hunk 3]
```diff
@@ -366,8 +370,9 @@ class Builder {
       this.onClickLocationCallback = this@Builder.onClickLocationCallback
       this.useGpsOnAddingLocation = this@Builder.useGpsOnAddingLocation
       this.mapLayers = this@Builder.mapLayers
-      this.shouldLocationButtonShow = this@Builder.shouldLocationButtonShow
+      this.showCurrentLocationButton = this@Builder.showCurrentLocationButton
       this.shouldPlaneSwitcherButtonShow = this@Builder.shouldPlaneSwitcherButtonShow
```
Comment (dubdabasoduba) [inline android/geowidget/src/main/java/org/smartregister/fhircore/geowidget/screens/GeoWidgetFragment.kt:374]:
```suggestion
      this.showPlaneSwitcherButton = this@Builder.showPlaneSwitcherButton
```

File: android/geowidget/src/main/java/org/smartregister/fhircore/geowidget/screens/GeoWidgetFragment.kt [hunk 4]
```diff
@@ -353,8 +356,9 @@ class Builder {
 
   fun setMapLayers(list: List<MapLayerConfig>) = apply { this.mapLayers = list }
 
-  fun setLocationButtonVisibility(show: Boolean) = apply { this.shouldLocationButtonShow = show }
+  fun showCurrentLocationButtonVisibility(show: Boolean) = apply { this.showCurrentLocationButton = show }
 
+  fun setAddLocationButtonVisibility(show: Boolean) = apply { this.showAddLocationButton = show }
   fun setPlaneSwitcherButtonVisibility(show: Boolean) = apply {
     this.shouldPlaneSwitcherButtonShow = show
```
Comment (dubdabasoduba) [inline android/geowidget/src/main/java/org/smartregister/fhircore/geowidget/screens/GeoWidgetFragment.kt:363]:
```suggestion
    this.showPlaneSwitcherButton = show
```

**Review by dubdabasoduba — CHANGES_REQUESTED** [review]
